# Data processing - Crawled Dataset (Beautiful Soup)

In [49]:
import pandas as pd
import re

In [36]:
dataset = pd.read_csv('/Users/sankeerthana/Documents/NTU/YEAR_4/SEM_2/CZ4034/IR-Sentiment-Analysis/IR-Sentiment-Analysis/skincare_dataset/productReviews_soup.csv')
dataset.head()

Product_ID          Product_Name Reviewer  \
0  6857050914885  Retinol Expert 0.1%   Caressa   
1  6857050914885  Retinol Expert 0.1%     Bryan   
2  6857050914885  Retinol Expert 0.1%   Tristan   
3  6857050914885  Retinol Expert 0.1%    Claire   
4  6857050914885  Retinol Expert 0.1%    Sun M.   

                                Review_Title  \
0                            The Real Deal!    
1                      Gave me a break-out?    
2                                              
3               Lovely and Doesn’t Irritate    
4   It's so great to have pure 0.1% retinol    

                                              Review  
0   I have been using this for 2 weeks now, and I...  
1   I received the sample size of this product an...  
2                                                NaN  
3                                                NaN  
4                                                NaN

### Data Exploration

In [7]:
dataset.shape

(62460, 5)

In [15]:
print("Number of Null Values:")
print(f"Review Title: {dataset['Review_Title'].isna().sum()}")
print(f"Review : {dataset['Review'].isna().sum()}")


Number of Null Values:
Review Title: 1
Review : 5256


The crawled dataset has 62460 rows and 5 columns out of which 5256 rows have null values.

In [41]:
dataset.drop(['Product_Name', 'Reviewer'], axis=1, inplace=True)
dataset

Product_ID                               Review_Title  \
0      6857050914885                            The Real Deal!    
1      6857050914885                      Gave me a break-out?    
2      6857050914885                                              
3      6857050914885               Lovely and Doesn’t Irritate    
4      6857050914885   It's so great to have pure 0.1% retinol    
...              ...                                        ...   
62455  3942309625925                                  Loved it    
62456  3942309625925                                  Soothing    
62457  3942309625925                  Great for sensitive skin    
62458  3942309625925                                       Wow    
62459  3942309625925              Super calming and hydrating!    

                                                  Review  
0       I have been using this for 2 weeks now, and I...  
1       I received the sample size of this product an...  
2                                                    NaN  
3                                                    NaN  
4                                                    NaN  
...                                                  ...  
62455                                          Loved it   
62456   This sheet mask soothes my acne prone skin. M...  
62457   I have very sensitive skin and this was great...  
62458   I got this for its calming effects and wasn't...  
62459   I love this mask! I’m a huge fan of sheet mas...  

[62460 rows x 3 columns]

In [50]:
for row in dataset.itertuples():
    print(re.search('(\s)+', row[2]))

 The Real Deal! 
 Gave me a break-out? 
 
 Lovely and Doesn’t Irritate 
 It's so great to have pure 0.1% retinol 
 The secret to smooth skin! 
 Love the way it feels! 
 Refreshing 
 Must-have 
 love it 
 Nice toner! 
 It’s a little bit to 
 Holy Grail 
 My Favorite Toner 
 Ph balanced toner 
 Decent 
 
 Favorite Toner 
 This has to always be in my daily routine 
 Tried and true! 
 Hello new favorite toner 
 Worlds Best Toner!! 
 Repurchased now at least 4 times! 
 Excellent toner! 
 Awesome! 
 Smells great and works even better! 
 Great Toner 
 Love this toner 
 Caused breakouts and red, irritated cheeks 
 
 Hydrating but a bit sticky 
 Best toner 
 Awesome Toner 
 good toner 
 Good, basic toner. 
 Staple toner 
 Great product 
 Love it 
 Solid 
 The best toner!!! 
 Immediately Brightens + refreshes skin 
 Best Toner 
 It's great 
 Great for combination skin 
 Best toner I’ve ever used 
 Oh how i’ve missed this product 
 Is the best product ever 
 My Rosacea Go To 
 Good 
 Best seller 

TypeError: expected string or bytes-like object

In [46]:
dataset.shape

(62458, 3)

In [35]:
dataset.head(n=20)

Product_ID                                 Review_Title  \
0   6857050914885                              The Real Deal!    
1   6857050914885                        Gave me a break-out?    
2   6857050914885                                                
3   6857050914885                 Lovely and Doesn’t Irritate    
4   6857050914885     It's so great to have pure 0.1% retinol    
5   6857051275333                  The secret to smooth skin!    
6   6857051275333                      Love the way it feels!    
7     10245844553                                  Refreshing    
8     10245844553                                   Must-have    
9     10245844553                                     love it    
10    10245844553                                 Nice toner!    
11    10245844553                        It’s a little bit to    
12    10245844553                                  Holy Grail    
13    10245844553                           My Favorite Toner    
14    10245844553                           Ph balanced toner    
15    10245844553                                      Decent    
16    10245844553                                                
17    10245844553                              Favorite Toner    
18    10245844553   This has to always be in my daily routine    
19    10245844553                             Tried and true!    

                                               Review  
0    I have been using this for 2 weeks now, and I...  
1    I received the sample size of this product an...  
2                                                 NaN  
3                                                 NaN  
4                                                 NaN  
5    This product 100% helps my skin feel smoother...  
6    I use this product as a toner in the mornings...  
7    Cleans off any dead skin but doesn’t leave my...  
8    This has been my go-to toner for nearly two y...  
9    i have used this product before and let me ju...  
10   It has a nice smell - kind of woodsy. It’s a ...  
11   It’s a little bit to strong for my skin after...  
12   Lost count how many times I’ve bought this, I...  
13   This toner feels hydrates and balances my ski...  
14   I cannot live without these toner a! It’s ama...  
15   It did the job but for some reason I never re...  
16                                                NaN  
17   Great for helping with red skin. Not drying. ...  
18   I have kept this in my routine for 2+ years! ...  
19   In the colder months this doesn’t hydrate me ...

In [40]:
dataset['Review'][16]

nan

In [47]:
dataset['Review'][16]

nan